slackbridge/messages.py

```diff
@@ -71,19 +83,54 @@ def resolve(self):
                 user_bot.leave(channel_name)
             return
 
+    def is_bot_user(self):
+        """Sometimes bot_id is not included and other
+        times it is passed as None. Checks both cases."""
+        return ('bot_id' in self.raw_message and
+                self.raw_message['bot_id'] is not None)
+
     def _change_presence(self, user_bot):
         if self.raw_message['presence'] == 'away':
             user_bot.away('Slack user inactive.')
         elif self.raw_message['presence'] == 'active':
             user_bot.back()
 
-    def _irc_me_action(self, channel_name, user_bot):
+    def _irc_me_action(self, channel_name, user_bot, action):
         user_bot.post_to_irc(
             user_bot.describe,
             '#' + channel_name,
-            self.raw_message['text'],
+            action,
         )
 
+    # TODO: Change image host to fluffy.cc in the future?
+    def _post_to_imgur(self, channel_name, user_bot, file_data):
+        ext = file_data['filetype']
+        if ext not in ALLOWED_FILE_EXTS:
+            return
+
+        auth = {'Authorization': 'Bearer {}'.format(
+            self.bridge_bot.slack_token)}
+        r = requests.get(file_data['url_private'],
+                         headers=auth, stream=True)
+        if r.status_code != 200:
+            log.msg('Could not GET image from: {}'.format(
+                file_data['url_private']))
+            return
+
+        with tempfile.NamedTemporaryFile('wb', suffix='.' + ext) as tempf:
+            r.raw.decode_content = True
+            shutil.copyfileobj(r.raw, tempf)
+
+            # The upload_image method calls raise_for_status
+            # which will trigger an HTTPError exception
+            try:
+                image = self.imgur.upload_image(tempf.name,
```

Comment:
I'd just make this `self.bridge_bot.imgur` so that you don't need to also have a attribute here in the message for it. Seems like that would make it a bit clearer where the imgur auth is coming from.